halo2_proofs/src/poly/commitment.rs

```diff
@@ -177,12 +177,23 @@ impl<C: CurveAffine> Params<C> {
         let k = u32::from_le_bytes(k);
         let n = 1 << k;
 
-        let g: Vec<C> = (0..n)
-            .map(|_| C::read(&mut reader))
-            .collect::<Result<_, _>>()?;
-        let g_lagrange: Vec<C> = (0..n)
-            .map(|_| C::read(&mut reader))
-            .collect::<Result<_, _>>()?;
+        let load_points_from_file_parallelly = |reader: &mut R| -> io::Result<Vec<C>> {
+            let mut points_compressed: Vec<C::Repr> = vec![C::Repr::default(); n];
+            for points_compressed in points_compressed.iter_mut() {
+                reader.read_exact((*points_compressed).as_mut())?;
+            }
+
+            let mut points = vec![C::default(); n];
+            parallelize(&mut points, |points, chunks| {
+                for (i, point) in points.iter_mut().enumerate() {
+                    *point = Option::from(C::from_bytes(&points_compressed[chunks + i])).unwrap();
+                }
+            });
+            Ok(points)
+        };
+
+        let g = load_points_from_file_parallelly(&mut reader)?;
+        let g_lagrange = load_points_from_file_parallelly(&mut reader)?;
```

Comment:
You could parrallelize these two calls too probably. Even with a thread or using async and polling the task.

I ignore if you have tried so. But I think it's worth the exploration.
I would start by calling 
```rust
        let g = load_points_from_file_parallelly(&mut reader)?;
        let g_lagrange = load_points_from_file_parallelly(&mut reader)?;
```
In sepparated threads and joining them instead to call them in series.
Also, would try to use async tasking  and join the futures. 

From the 3 options then, keep the one that performs best.

WDYT @spherel ? :smile:

Reply:
Although i don't have real bench results, in this pr we already use every cpu cores during decoding points, which is "heavier" that reading bytes from disk. And paralleling these two loading will not benefit cpu usage. I high doubt it will bring very small benefit while making codes less easier for reading

Reply:
Hmmm for the decoding, I think the easiest will be to simply create an `unsafe_read` fn in `pairing` crate which does no checks. On that way, the "decoding" efforts even parallelized, would not deal with many stuff.

Let me file an issue for that.

Reply:
For the rest makes complete sense. Re-reviewing to approve! :)